projects/plugins/jetpack/modules/masterbar/profile-edit.php

```diff
@@ -0,0 +1,56 @@
+<?php
+/**
+ * WP-Admin Profile edit.
+ *
+ * @package Jetpack
+ */
+
+use Automattic\Jetpack\Connection\Manager as Connection_Manager;
+
+/**
+ * Hides profile fields for WordPress.com connected users.
+ *
+ * @param WP_User $user The current WP_User object.
+ */
+function masterbar_hide_profile_fields( $user ) {
+	$user_id            = $user->ID;
+	$connection_manager = new Connection_Manager( 'jetpack' );
+	if ( ! $connection_manager->is_user_connected( $user_id ) ) {
```

Comment:
Nitpick: I think using directly $user->ID is pretty explicit, I don't think we need an additional variable. :)

Reply:
Good idea 71f6f02